pkg/tenant/projects.go

```diff
@@ -10,29 +10,68 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
+// TenantProjectList retrieves all projects assigned to a tenant
+// and returns a 200 OK response.
+//
+// Route: //tenant/:tenantID/projects
 func (s *Server) TenantProjectList(c *gin.Context) {
-	c.JSON(http.StatusNotImplemented, "not implemented yet")
+	var (
+		err error
+	)
+	// Get the project's tenant ID from the URL and return a 400 response
+	// if the tenant ID is not a ULID.
+	var tenantID ulid.ULID
+	if tenantID, err = ulid.Parse(c.Param("tenantID")); err != nil {
+		log.Error().Err(err).Msg("could not parse tenant ulid")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not parse tenant ulid"))
+		return
+	}
+
+	// Get projects from the database and return a 500 response
+	// if not successful.
+	if _, err := db.ListProjects(c.Request.Context(), tenantID); err != nil {
+		log.Error().Err(err).Msg("could not fetch projects from the database")
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not fetch projects from the database"))
+		return
+	}
+
+	// Build the response.
+	out := &api.TenantProjectPage{TenantProjects: make([]*api.Project, 0)}
+
+	tenantProject := &api.Project{}
+
+	out.TenantProjects = append(out.TenantProjects, tenantProject)
+
+	c.JSON(http.StatusOK, out)
 }
 
 func (s *Server) TenantProjectCreate(c *gin.Context) {
 	c.JSON(http.StatusNotImplemented, "not implemented yet")
 }
 
+// ProjectList retrieves all projects assigned to a tenant
+// and returns a 200 OK response.
+//
+// Route: /projects
```

Comment:
Again, I'm not totally sure I get the distinction between `TenantProjectList` and `ProjectList`, does it have to do with the specificity of the client's request? Or maybe it's a use case thing, e.g. if I have multiple tenants vs. if I only have one tenant? Perhaps we can add a bit of clarification in the docstring?

Reply:
We want `ProjectList` to provide a list of projects for an organization and `TenantProjectList` will list projects on the tenant. I'll update the docstring.